Client/Application/LeanplumIntegration.swift

```diff
@@ -0,0 +1,201 @@
+        }
+
+        guard let settings = getSettings() else {
+            log.error("LeanplumIntegration - Could not load settings from Info.plist")
+            return
+        }
+
+        switch settings.environment {
+        case .development:
+            log.info("LeanplumIntegration - Setting up for Development")
+            Leanplum.setDeviceId(UIDevice.current.identifierForVendor?.uuidString)
+            Leanplum.setAppId(settings.appId, withDevelopmentKey: settings.key)
+        case .production:
+            log.info("LeanplumIntegration - Setting up for Production")
+            Leanplum.setAppId(settings.appId, withProductionKey: settings.key)
+        }
+        Leanplum.syncResourcesAsync(true)
+        setupTemplateDictionary()
+
+        var userAttributesDict = [AnyHashable: Any]()
+        userAttributesDict["Alternate Mail Client Installed"] = "mailto:"
+        userAttributesDict["Focus Installed"] = canInstallFocus()
+        userAttributesDict["Klar Installed"] = canInstallKlar()
+
+        if let mailtoScheme = self.profile?.prefs.stringForKey(PrefsKeys.KeyMailToOption), mailtoScheme != "mailto:" {
+            userAttributesDict["Alternate Mail Client Installed"] = mailtoScheme
+        }
+
+        Leanplum.start(userAttributes: userAttributesDict)
+
+        Leanplum.track(LeanplumEventName.openedApp.rawValue)
+    }
+
+    // Events
+
+    func track(eventName: LeanplumEventName) {
+        if shouldSendToLP() {
+            Leanplum.track(eventName.rawValue)
+        }
+    }
+
+    func track(eventName: LeanplumEventName, withParameters parameters: [String: AnyObject]) {
+        if shouldSendToLP() {
+            Leanplum.track(eventName.rawValue, withParameters: parameters)
+        }
+    }
+
+    // States
+
+    func advanceTo(state: String) {
+        if shouldSendToLP() {
+            Leanplum.advance(to: state)
+        }
+    }
+
+    // Data Modeling
+
+    func setupTemplateDictionary() {
+        if shouldSendToLP() {
+            LPVar.define("Template Dictionary", with: ["Template Text": "", "Button Text": "", "Deep Link": "", "Hex Color String": ""])
+        }
+    }
+
+    func getTemplateDictionary() -> [String:String]? {
+        if shouldSendToLP() {
+            return Leanplum.object(forKeyPathComponents: ["Template Dictionary"]) as? [String : String]
+        }
+        return nil
+    }
+
+    func getBoolVariableFromServer(key: String) -> Bool? {
+        if shouldSendToLP() {
+            return Leanplum.object(forKeyPathComponents: [key]) as? Bool
+        }
+        return nil
+    }
+
+    // Utils
+    
+    func setEnabled(_ enabled: Bool) {
+        self.enabled = enabled
+        if self.enabled {
+            self.start()
+        }
+    }
+
+    func canInstallFocus() -> Bool {
+        guard let focus = URL(string: "focus://") else {
+            return false
+        }
+        return !UIApplication.shared.canOpenURL(focus)
+    }
+
+    func canInstallKlar() -> Bool {
```

Comment:
Ah, I just added a ! to flip the logic to be correct. Let me know if this is a must.